src/main/java/com/cegeka/academy/service/invitation/ValidationAccessService.java

```diff
@@ -0,0 +1,45 @@
+package com.cegeka.academy.service.invitation;
+
+import com.cegeka.academy.domain.User;
+import com.cegeka.academy.repository.InvitationRepository;
+import com.cegeka.academy.service.UserService;
+import com.cegeka.academy.web.rest.errors.UnauthorizedUserException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+@Transactional
+public class ValidationAccessService {
+
+    private final UserService userService;
+    private final InvitationRepository invitationRepository;
+
+    @Autowired
+    public ValidationAccessService(UserService userService, InvitationRepository invitationRepository) {
+        this.userService = userService;
+        this.invitationRepository = invitationRepository;
+    }
+
+    public boolean verifyUserAccessForInvitationEntity(Long invitationId){
+
+        Optional<User> opt = userService.getUserWithAuthorities();
+        final User[] userLogged = {null};
+        opt.ifPresent(user -> userLogged[0] = user);
+
+        User invitedUser = invitationRepository.findById(invitationId).get().getUser();
```

Comment:
Ce se intampla daca nu primesti un id de invitatie care sa existe in baza?
Arunca un ochi [aici](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--).